maths/radix2_fft.py

```diff
@@ -0,0 +1,202 @@
+                    )
+                current_root *= root
+            # Second half of next step
+            current_root = 1
+            for j in range(
+                self.C_max_length // (next_ncol * 2)
+            ):
+                for i in range(next_ncol):
+                    new_dft[i].append(
+                        dft[i][j]
+                        - current_root
+                        * dft[i + next_ncol][j]
+                    )
+                current_root *= root
+            # Update
+            dft = new_dft
+            next_ncol = next_ncol // 2
+        return dft[0]
+
+    # multiply the DFTs of  A and B and find A*B
+    def __multiply(self):
+        dftA = self.__DFT("A")
+        dftB = self.__DFT("B")
+        inverseC = [
+            [
+                dftA[i] * dftB[i]
+                for i in range(self.C_max_length)
+            ]
+        ]
+        del dftA
+        del dftB
+
+        # Corner Case
+        if len(inverseC[0]) <= 1:
+            return inverseC[0]
+        # Inverse DFT
+        next_ncol = 2
+        while next_ncol <= self.C_max_length:
+            new_inverseC = [[] for i in range(next_ncol)]
+            root = self.root ** (next_ncol // 2)
+            current_root = 1
+            # First half of next step
+            for j in range(self.C_max_length // next_ncol):
+                for i in range(next_ncol // 2):
+                    # Even positions
+                    new_inverseC[i].append(
+                        (
+                            inverseC[i][j]
+                            + inverseC[i][
+                                j
+                                + self.C_max_length
+                                // next_ncol
+                            ]
+                        )
+                        / 2
+                    )
+                    # Odd positions
+                    new_inverseC[i + next_ncol // 2].append(
+                        (
+                            inverseC[i][j]
+                            - inverseC[i][
+                                j
+                                + self.C_max_length
+                                // next_ncol
+                            ]
+                        )
+                        / (2 * current_root)
+                    )
+                current_root *= root
+            # Update
+            inverseC = new_inverseC
+            next_ncol *= 2
+        # Unpack
+        inverseC = [
+            round(x[0].real, 8) + round(x[0].imag, 8) * 1j
+            for x in inverseC
+        ]
+
+        # Remove leading 0's
+        while inverseC[-1] == 0:
+            inverseC.pop()
+        return inverseC
+
+    # Overwrite __str__ for print(); Shows A, B and A*B
+    def __str__(self):
+        A = "A = "
+        B = "B = "
+        C = "A*B = "
+        for i in range(self.len_A):
+            A += str(self.polyA[i]) + "*x^" + str(i) + " + "

```

Comment:
A += f"{self.polyA[i]}*x{i} + "

Reply:
I updated the `__str__ ` method using f prefix for clarity and list comprehension+String.join() for performance.